sso/test-steps/src/main/java/org/eclipse/kapua/service/sso/steps/SsoServiceSteps.java

```diff
@@ -0,0 +1,211 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SsoServiceSteps.class);
+    private static final String KEYCLOAK_TOKEN_REST_URI = "http://localhost:9090/auth/realms/kapua/protocol/openid-connect/token";
+    private static JwtAuthenticatingRealm jwtAuthenticatingRealm;
+    public static SecurityManager securityManager;
+    private CredentialsFactory credentialsFactory;
+    private AuthenticationService authenticationService;
+
+    // Default constructor
+    @Inject
+    public SsoServiceSteps(StepData stepData, DBHelper dbHelper) {
+
+        this.stepData = stepData;
+        this.database = dbHelper;
+    }
+
+    // ************************************************************************************
+    // * Setup and tear-down steps                                                        *
+    // ************************************************************************************
+
+    @Before
+    public void beforeScenario(Scenario scenario) throws Exception {
+
+        this.scenario = scenario;
+        database.setup();
+        stepData.clear();
+
+        locator = KapuaLocator.getInstance();
+        credentialsFactory = locator.getFactory(CredentialsFactory.class);
+        authenticationService = locator.getService(AuthenticationService.class);
+
+        // Setup JAXB context
+        XmlUtil.setContextProvider(new TestJAXBContextProvider());
+    }
+
+    @After
+    public void afterScenario() {
+
+        // Clean up the database
+        try {
+            LOGGER.info("Logging out in cleanup");
+            if (isIntegrationTest()) {
+                database.deleteAll();
+                SecurityUtils.getSubject().logout();
+            } else {
+                database.dropAll();
+                database.close();
+            }
+            KapuaSecurityUtils.clearSession();
+        } catch (Exception e) {
+            LOGGER.error("Failed to log out in @After", e);
+        }
+    }
+
+    // ************************************************************************************
+    // * Definition of Cucumber scenario steps                                            *
+    // ************************************************************************************
+
+    @And("^Set environment variables for keycloak provider$")
+    public void setEnvVariablesForKeycloakProvider() throws UnknownHostException {
+        InetAddress address = InetAddress.getLocalHost();
+//        System.setProperty("sso.keycloak.uri", address.getHostAddress() + ":9090");
+        System.setProperty("sso.keycloak.uri", "http://localhost:9090");
+        System.setProperty("site.home.uri", "http://localhost:8080");
+        System.setProperty("sso.keycloak.realm", "kapua");
+        System.setProperty("sso.openid.client.id", "console");
+        System.setProperty("sso.provider", "keycloak");
+        System.setProperty("authentication.registration.service.enabled", "true");
+    }
```

Comment:
I would prefer setting the `authentication.registration.service.enabled` property as `true` only for the scenarios that actually use the `RegistrationServiceImpl`, otherwise this kind of registration is enabled for each scenario.

Reply:
This should now be resolved. I added this line to the other method so it is only used when `RegistrationServiceImpl `is used.

Reply:
Hi @zanpizmoht , ok. Just a question concerning the `configureTheSso()` method, I saw that this is used also when the `RegistrationService` (or the `SimpleRegistrationProcessor`) is not mentioned in the scenario description. Is it because this kind of registration is used also for other tests?

Reply:
Which scenario do you have in mind?

Reply:
For instance _'Using normal login using Keycloak credentials Create user using keycloak Login as external user using normal login method Exception should be thrown'_, _'Logging out and logging back in with SSO Login as external user, logout Login again, no exception should be thrown'_ and _'Login using normal login after logging out with SSO Login as external user, logout Login again, this time with normal login method, exception should be thrown'_ . Do these scenarios also require the `RegistrationServiceImpl` ?

Reply:
Yes both scenarios are requiring the `RegistrationServiceImpl` class because they use external user login.

Reply:
But the `RegistrationServiceImpl` should be needed only for testing the `SimpleRegistrationProcessor` (automatic creation of the user in Kapua at the first log-in attempt). In the other cases we would like to test the external user login with existing (external) users, and such login method shouldn't require the `RegistrationServiceImpl`.

Reply:
Yes I know, this is also covered in the scenario 'Create external user using the the UserService'. This is just the basic scenario, I will add more when this PR is merged.

Reply:
Ok. If you are using the `RegistrationServiceImpl` also in the scenarios above, it would be better to mention that the external user is created via `SimpleRegistrationProcessor`, like you did in another scenario. Since the `RegistrationServiceImpl` can trigger the creation of new users in Kapua that can login even if they didn't exist before (I guess this is the case for the test scenario _'Logging out and logging back in with SSO Login as external user, logout Login again, no exception should be thrown'_), I would prefer to explicitly state when it is used. In this way we will be able to distinguish these scenarios from the ones that won't use the `RegistrationServiceImpl`.

Reply:
You are right, I will add this to the descriptions of those scenarios.